### DISCUSSION TITLE
Player to display media on second screen

### DISCUSSION THREAD

**Discussion by SandroCODTEC** [sircharlo/meeting-media-manager#218] [Ideas] [answered]
Congratulations for the excellent work.  Now, with the return of face-to-face meetings, would it be possible to implement a player to display the media on the second screen?

**Comment by sircharlo** [comment, marked as answer]
Yes indeed; I'm cooking something up as we speak. 

Features I'm implementing:
- Easy selection of meeting and media to display
- Prevent accidental closing of JWMMF while media is being shown
- Configurable default background (to allow for yeartext to be displayed, for example)

Hoping to release a preliminary "beta" relatively soon to get some feedback.

**Reply by SandroCODTEC** [reply]
What excellent news.
Another feature that would be very useful and that existed in the Sandbox was the possibility to control the media through the smartphone.
1. The App does a get on host:port and authenticates with the explicit permission of JWMMF;
2. The App receives the list of media and could display and control the content (images, videos and audios);

Extremely useful resource in small congregations. Also very useful if the speaker wants to control the media to be displayed in his speech.

**Reply by Uilsonjw** [reply]
It's very nice have this option to play media on  second monitor. On my tests it's the easy way to hybrid meetings, however would be nice if the software is able to run even without internet connection. Of course if the connection fails we will not able to run zoom but we still able to play the media in the kingdom hall.

**Reply by SandroCODTEC** [reply]
The ability to control the media by external software (an Android app for example) is a feature that does not prevent the system from working offline.  In fact, it is even preferable that the connection occurs on the same network, informing the IP: Port of the machine that runs the system.

**Reply by Uilsonjw** [reply]
Sorry, I just suggested to permit JW Media Feather to work offline to be able to play the media in the kingdom hall even without internet. I'm not concern about remote controller.

**Reply by Uilsonjw** [reply]
I just imagined that here is a good place to post this request. Do you think that I have to put it on another place?

**Reply by joaochris** [reply]
> I just imagined that here is a good place to post this request. Do you think that I have to put it on another place?

@Uilsonjw This is a great idea! Really if we run out of internet the program doesn't open. It would be nice if it works offline. You can create a discussion for this in the same tab, choose the ideas category 💡 and go to New discussion.

**Reply by sircharlo** [reply]
> Another feature that would be very useful and that existed in the Sandbox was the possibility to control the media through the smartphone.
> 
> 1. The App does a get on host:port and authenticates with the explicit permission of JWMMF;
> 
> 2. The App receives the list of media and could display and control the content (images, videos and audios);
> 
> 
> 
> Extremely useful resource in small congregations. Also very useful if the speaker wants to control the media to be displayed in his speech.

This might be a bit more involved to implement. I'll see if that's feasible at some point in the future.

**Reply by sircharlo** [reply]
> It's very nice have this option to play media on  second monitor. On my tests it's the easy way to hybrid meetings, however would be nice if the software is able to run even without internet connection. Of course if the connection fails we will not able to run zoom but we still able to play the media in the kingdom hall.
> 
> 

Very true. JWMMF started when meetings were on Zoom, therefore a meeting with no internet wasn't really possible! Now things are changing. I'll update the logic flow to allow for situations where the internet is down but we still need to show media. Thanks for the idea!
